samza-sql/src/main/java/org/apache/samza/sql/interfaces/SamzaRelConverter.java

```diff
@@ -0,0 +1,78 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.samza.sql.interfaces;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.schema.Table;
+import org.apache.calcite.schema.TableFactory;
+import org.apache.samza.sql.data.SamzaSqlRelMessage;
+
+
+/**
+ * Samza SQL application uses {@link SamzaRelConverter} to convert the Samza messages to relational message before
+ * it can be processed by the calcite engine.
+ * The {@link SamzaRelConverter} is configurable at a system level, So it is possible to configure different
```

Comment:
Are you saying we should be able to support stream level overrides or We should just move to stream level relconverter and not have system level relConverter? If it is later, We cannot because if you look at SamzaSQL as a service, it will not know all the streams it needs to support. 

If you are recommending we should support stream level overrides, We could definitely add that capability, Just that i haven't found a real use for that capability yet in our usecases.

Reply:
As discussed in our meeting, we will add the stream level overrides in future if there is a need for it.